app/routes/admin.py

```
@@ -153,28 +151,117 @@ def update_schedule(user_id):
 
     return redirect(url_for("admin.admin_dashboard"))
+
+@admin.route("/admin_dashboard/manage_admin/<int:user_id>", methods=["GET", "POST"])
+@login_required
+@admin_required
+def manage_admin(user_id):
+    user = User.query.get_or_404(user_id)
+
+    # Prevent self-demotion from admin role
+    if current_user.id == user.id and request.method == "POST":
+        if request.form.get("role") != "admin":
```

Ruff (0.8.2)

Use a single `if` statement instead of nested `if` statements

(SIM102)